include/seastar/coroutine/exception.hh

```diff
@@ -104,9 +111,16 @@ exception make_exception(T&& t) {
 /// ```
 /// co_await coroutine::return_exception(std::runtime_error("something failed miserably"));
 /// ```
+template<typename T>
+requires std::same_as<std::remove_reference_t<T>, std::exception_ptr>
+[[nodiscard]]
+exception return_exception(T&& t) noexcept {
```

Comment:
It's not good form to have an API accept both std::exception_ptr and a value type and treat them differently.

Reply:
Do you suggest to get only `std::exception_ptr` and let the caller make it if needed?
We can do that, but I'm afraid that will require to bump the api version.

Reply:
Also, do we want to present narrower API than make_exception_future(), or for the sake of symmetry we should keep up with the badly formed API, which I don't find so bad by the way.
To me it feels like many other polymorphic interfaces in the c++ standard library that accept different input types and treat them differently.  I don't really understand why here you find it so distasteful.